### PR TITLE
Add libinjection check on last path segment

### PR DESCRIPTION
**Pull request by dune73**
This is a re-implementation of PR #1329 by @theseion. It addresses two open issues:

* User REQUEST_BASENAME
* add tests

### CODE REVIEW

File: rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf [hunk 1]
```diff
@@ -1504,6 +1506,41 @@ SecRule ARGS "@rx \W{4}" \
...
+    tag:'language-multi',\
+    tag:'platform-multi',\
+    tag:'attack-sqli',\
+    tag:'OWASP_CRS/WEB_ATTACK/SQL_INJECTION',\
+    tag:'WASCTC/WASC-19',\
+    tag:'OWASP_TOP_10/A1',\
+    tag:'OWASP_AppSensor/CIE1',\
+    tag:'PCI/6.5.2',\
+    tag:'paranoia-level/3',\
+    ver:'OWASP_CRS/3.1.0',\
```
Comment (fgsch) [inline rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:1539]: Do we want 3.2.0 here?

**Comment by fgsch** [comment]
Fixes #1328

**Comment by fgsch** [comment]
LGTM 👍

File: rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf (hunk 1 above)
Comment (dune73) [reply rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf:1539]: It's 3.1.0 across the board. I take it we are going to script-update it before 3.2.0-rc1.

**Comment by dune73** [comment]
Thank you for your review.

**Pull request merged**